packages/vite/src/node/server/middlewares/error.ts

```diff
@@ -77,10 +77,14 @@ export function errorMiddleware(
             <meta charset="UTF-8" />
             <title>Error</title>
             <script type="module">
-              import { ErrorOverlay } from '/@vite/client'
-              document.body.appendChild(new ErrorOverlay(${JSON.stringify(
-                prepareError(err),
-              ).replace(/</g, '\\u003c')}))
+              try {
+                const { ErrorOverlay } = await import('/@vite/client')
+                document.body.appendChild(new ErrorOverlay(${JSON.stringify(
+                  prepareError(err),
+                ).replace(/</g, '\\u003c')}))
+              } catch {
+                document.innerHTML = \`<h1>Internal Server Error</h1><h2>Error overlay failed to load</h2><pre>${err.stack?.replaceAll('<', '&lt;').replaceAll('$', '&dollar;')}</pre>\`
+              }
```

Comment:
Question: Could we have the `JSON.stringify(prepareError(err)).replace(...)` kept in a variable outside the try-block, and re-use it here too? That way we have a consistent formatting of the error (e.g. stacktrace is ansi-stripped), and we don't need to escape here differently.

Reply:
I'd originally tried doing the same `.replace(/</g, '\\u003c')`, but it wasn't working. I don't fully understand why it needs different escaping here or the difference between them. In any case, we would still need an extra escaping of `$` in this second use because it's being used as a template string, which is different than the first use